src/Microsoft.ML.Data/DataDebuggerPreview.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// This class represents an eager 'preview' of a <see cref="IDataView"/>.
+    /// </summary>
+    public sealed class DataDebuggerPreview
+    {
+        internal static class Defaults
+        {
+            public const int MaxRows = 100;
+        }
+
+        public Schema Schema { get; }
+        public ColumnInfo[] ColumnView { get; }
+        public RowInfo[] RowView { get; }
+
+        internal DataDebuggerPreview(IDataView data, int maxRows = Defaults.MaxRows)
+        {
+            Contracts.CheckValue(data, nameof(data));
+            Contracts.CheckParam(maxRows >= 0, nameof(maxRows));
+            Schema = data.Schema;
+
+            int n = data.Schema.ColumnCount;
+
+            var rows = new List<RowInfo>();
+            var columns = new List<object>[n];
+            for (int i = 0; i < columns.Length; i++)
+                columns[i] = new List<object>();
+
+            using (var cursor = data.GetRowCursor(c => true))
```

Comment:
Do we have any concern when using an MLContext that has `conc` set to a value other than `1`? I know that in the past when this occurs, the Debugger requires "force all threads to execute" in order to get these values.  I'm not sure how well it works in other IDEs - like VS Code and VS for Mac.

Reply:
You still have to execute all threads. I'm not sure how to improve that story.

---
In reply to: [230910664](https://github.com/dotnet/machinelearning/pull/1518#discussion_r230910664) [](ancestors = 230910664)